### ISSUE TITLE
Add extensively tuned XGboost and Catboost baseline

### ISSUE THREAD

**Issue by yiweny** [pyg-team/pytorch-frame#53]
After discussing with @weihua916 in office today, we decided to include extensively tuned XGboost and Catboost baseline, so users of the pytorch-frame can easily compare the performance of the deep learning approaches to GBDT approaches. 

The idea is inspired by ExcelFormer as the authors compared the performance of ExcelFormer with extensively tuned XGboost and Catboost. 

The intented interface looks like the following:

```python
xgboost = torch_frame.XGBoost()
xgboost.fit_tune(tf_train, tf_val, num_trials, task_type, task_kwargs, search_kwargs)
y_pred_test = xgboost.predict(tf_test)
test_acc = accuracy(y_pred_test, tf_test.y)
```
The `fit_tune` function will do hyper parameter search by default. If `num_trials` is 1, then we don't do any hyperparameter search and use the default hyper-parameter values. 
`task_kwargs` is any keyword arguments for the task (e.g., number of classes for classification, MAE versus RMSE loss for regression). 
`search_kwargs` is any keyword arguments to manually specify the hyper-parameter search space.

Here is the default hyper parameter tuning space in ExcelFormer and we propose it as the default search space for our `XGBoost` and `Catboost` baselines. 
<img width="397" alt="Screenshot 2023-09-05 at 10 46 11 PM" src="https://github.com/pyg-team/pytorch-frame/assets/22352182/e5178b75-ea7e-4294-aced-2dc111b2be1a">
